azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,499 @@
+package subscription
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-11-01/subscriptions"
+	subscriptionAlias "github.com/Azure/azure-sdk-for-go/services/subscription/mgmt/2020-09-01/subscription"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	azValidate "github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/subscription/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/subscription/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var (
+	billingScopeEnrollmentFmt = "/providers/Microsoft.Billing/billingAccounts/%s/enrollmentAccounts/%s"
+	billingScopeMCAFmt        = "/providers/Microsoft.Billing/billingAccounts/%s/billingProfiles/%s/invoiceSections/%s"
+)
+
+func resourceSubscription() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSubscriptionCreate,
+		Update: resourceSubscriptionUpdate,
+		Read:   resourceSubscriptionRead,
+		Delete: resourceSubscriptionDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
```

Comment:
can we make this:

```suggestion
		Importer: azSchema.ValidateResourceIDPriorToImportThen(func(id string) error {
```

and then validate the subscription isn't cancelled during creation